### ISSUE TITLE
Github Groups not populated on account sign-in

### ISSUE THREAD

**Issue by unsubtleguy** [pomerium/pomerium#3160]
## What happened?

When using pomerium latest, using helm chart 30.1.1 and github as an identity provider, user groups are not populated, resulting in routes requiring group membership to return a 403 forbidden.

After an amount of time (somewhere between minutes to tens of minutes), group memberships are populated, allowing route checks to pass.

Group memberships when hitting the `authenticate` route showing user info:

![image](https://user-images.githubusercontent.com/19356391/158902821-4df45da0-f0da-4dcb-a626-84bad2213298.png)

## What did you expect to happen?

When signing in, group memberships should be populated, and users should not have to wait minutes before being allowed to reach routes requiring group memberships.

## How'd it happen?

1. Have an unauthenticated session
2. Browse to a route requiring group memberships (i.e. `allowed_groups` with a github group of owners)
3. Get redirected to github, click 'allow'/etc, get redirected to route
4. get a 403 message

Wait a couple of minutes, refresh the page, now groups work and user can access protected resource.

## What's your environment like?

- Pomerium version (retrieve with `pomerium --version`):
```
$ k -n pomerium exec pomerium-authenticate-XXXXX-YYYYY -- pomerium --version
pomerium: 0.16.4-1645833281+0d3fa003
envoy: 1.19.3+0799f0751367f33f54e488ff1241104bb9592a916e5c2387d80776291f8fed45

```
- Server Operating System/Architecture/Cloud:

* Kubernetes 1.23.5
* Using helm chart version 30.1.1
* Not using redis, tested with both redis and memory backed store, both exhibit the same behaviour

## What's your config.yaml?

```config.yaml
authenticate:
  existingTLSSecret: pomerium-tls-internal
  idp:
    provider: "github"
    clientID: <snippy>
    clientSecret: "<Snipped>"
    # Misc - This personal access token has `read:org, user:email` rights
    serviceAccount: "<base64'd json blob of form {'username:'user','personal_access_token:'fooo'}>"
  proxied: true
  # This is authenticate.ingress - not to be confused with the next entry
  ingress:
    tls:
      # If all else fails, this is the one you want to modify
      secretName: pomerium-tls-external
# ingress, not authenticate.ingress!
ingress:
  hosts:
    - "*.sample-domain.com"
  tls:
    hosts:
      - "*.sample-domain.com"
  secretName: pomerium-tls-external

proxy:
  existingTLSSecret: pomerium-tls-internal
  deployment:
    # Work around the small (30s) default time for clients to send data
    extraEnv:
      TIMEOUT_READ: "600s"

databroker:
  existingTLSSecret: pomerium-tls-internal
  storage:
    #connectionString: rediss://pomerium-redis-master.pomerium.svc.cluster.local
    type: memory
    clientTLS:
      existingSecretName: pomerium-redis-tls
      existingCASecretKey: ca.crt

authorize:
  existingTLSSecret: pomerium-tls-internal

redis:
  enabled: false
  generateTLS: false
  tls:
    certificateSecret: pomerium-redis-tls

ingressController:
  enabled: true

config:
  rootDomain: sample-domain.com
  existingCASecret: pomerium-tls-internal
  generateSigningKey: false
  existingSigningKeySecret: pomerium-signing-key
  generateTLS: false
  routes:
    - from: https://argo.prod.sample-domain.com
      to: https://argocd-server.argocd.svc.cluster.local:443
      tls_skip_verify: true
      allowed_groups:
        - owners
```

## What did you see in the logs?

```logs
$k -n pomerium logs pomerium-proxy-xxx-yyy
<snipped, but basically other normal user traffic>
{"level":"info","service":"envoy","upstream-cluster":"","method":"GET","authority":"argo.prod.sample-domain.com","path":"/","user-agent":"Mozilla/5.0 (X11; Fedora; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36","referer":"","forwarded-for":"192.168.10.196","request-id":"6ab3d226-9092-41ef-8f3f-f693d4cef31d","duration":10.033618,"size":11824,"response-code":403,"response-code-details":"ext_authz_denied","time":"2022-03-17T22:21:31Z","message":"http-request"}


k -n pomerium logs pomerium-databroker-xxx-yyy -f 
{"level":"info","type":"type.googleapis.com/user.User","id":"unsubtleguy","time":"2022-03-17T22:20:41Z","message":"get"}
{"level":"info","type":"type.googleapis.com/user.User","id":"unsubtleguy","time":"2022-03-17T22:20:42Z","message":"put"}
{"level":"info","type":"type.googleapis.com/session.Session","id":"d78f2a6a-c73a-4954-a9b1-82ede0448808","time":"2022-03-17T22:20:42Z","message":"put"}
{"level":"info","type":"type.googleapis.com/directory.User","id":"unsubtleguy","time":"2022-03-17T22:20:43Z","message":"put"}
{"level":"info","type":"type.googleapis.com/session.Session","id":"d78f2a6a-c73a-4954-a9b1-82ede0448808","time":"2022-03-17T22:20:43Z","message":"get"}
{"level":"info","type":"type.googleapis.com/session.Session","id":"d78f2a6a-c73a-4954-a9b1-82ede0448808","time":"2022-03-17T22:20:44Z","message":"get"}
{"level":"info","type":"type.googleapis.com/user.User","id":"unsubtleguy","time":"2022-03-17T22:20:44Z","message":"get"}
{"level":"info","type":"type.googleapis.com/directory.User","id":"unsubtleguy","time":"2022-03-17T22:20:44Z","message":"get"}
{"level":"info","type":"type.googleapis.com/directory.Group","id":"","time":"2022-03-17T22:20:44Z","message":"get"}
{"level":"info","type":"type.googleapis.com/session.Session","id":"4a52d8c8-9ae2-4885-8805-ac050a1ad4ec","time":"2022-03-17T22:20:55Z","message":"get"}
{"level":"info","type":"type.googleapis.com/session.Session","id":"d78f2a6a-c73a-4954-a9b1-82ede0448808","time":"2022-03-17T22:20:55Z","message":"get"}


k -n pomerium logs pomerium-authorize-xxx-yyy
{"level":"info","service":"authorize","request-id":"d05c94db-b410-4c0a-be07-13f3b9cd4e9d","check-request-id":"93d8480e-42e6-4539-a170-8629dd6ea2af","method":"GET","path":"/","host":"argo.prod.sample-domain.com,"query":"","session-id":"d78f2a6a-c73a-4954-a9b1-82ede0448808","allow":false,"allow-why-false":["groups-unauthorized","non-pomerium-route"],"deny":false,"deny-why-false":["valid-client-certificate-or-none-required"],"user":"unsubtleguy","email":"github@domain.com","databroker_server_version":15255767957071690341,"databroker_record_version":1608,"time":"2022-03-17T22:24:50Z","message":"authorize check"}

```

The event which seems to 'resolve' the issue is as follows from the databroker:

```

{"level":"info","service":"identity_manager","time":"2022-03-17T22:29:03Z","message":"refreshing directory users"}

```
After this point, group memberships are populated and are working. From logs, this seems to run every 10 minutes.

## Additional context

* Basically a stock helm setup, using github as an IDP
* Personal access token works for fetching group memberships, as eventually groups are fetched
* Swapping in/out redis doesn't seem to have an effect
* It looks like on login, github groups are not being fetched
* At some point, a periodic job runs and then groups are retrieved, then authentication works

**Comment by calebdoxsey** [comment]
@unsubtleguy this was caused by a bug in our directory sync code. We weren't querying the GraphQL API properly. I have a PR to fix it.

**Comment by unsubtleguy** [comment]
Awesome - thanks for that @calebdoxsey

**Comment by unsubtleguy** [comment]
Can confirm, updated helm chart to use the latest image tagged main, works as intended

Thanks again!
